Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -203,4 +294,22 @@ Vector<StyledMarkedText> StyledMarkedText::coalesceAdjacentWithEqualDecorations(
     });
 }
 
+void StyledMarkedText::removeOverlappingForegroundRanges(Vector<StyledMarkedText>& markedTexts)
+{
+    HashSet<std::pair<unsigned, unsigned>> ranges;
+    for (auto& markedText : markedTexts) {
+        if (!markedText.highlightName.isNull())
+            ranges.add(std::make_pair(markedText.startOffset, markedText.endOffset));
+    }
+    // Check for non-highlight StyledMarkedTexts that overlap, if so then remove them.
+    unsigned index = 0;
+    while (index < markedTexts.size()) {
+        if (markedTexts[index].highlightName.isNull() && ranges.contains(std::make_pair(markedTexts[index].startOffset, markedTexts[index].endOffset))) {
+            markedTexts.remove(index);
+            continue;
+        }
+        index++;
```

Comment:
Although it doesn't matter for built-in types, it is good practice to use pre-increment / pre-decrement whenever possible (instead of the postfix version).